subprojects/core/src/main/java/org/gradle/caching/configuration/internal/BuildCacheConfigurationInternal.java

```diff
@@ -19,10 +19,23 @@
 import org.gradle.caching.BuildCacheServiceFactory;
 import org.gradle.caching.configuration.BuildCache;
 import org.gradle.caching.configuration.BuildCacheConfiguration;
+import org.gradle.caching.local.DirectoryBuildCache;
+
+import javax.annotation.Nullable;
 
 public interface BuildCacheConfigurationInternal extends BuildCacheConfiguration {
     /**
      * Finds a build cache implementation factory class for the given configuration type.
      */
     <T extends BuildCache> Class<? extends BuildCacheServiceFactory<T>> getBuildCacheServiceFactoryType(Class<T> configurationType);
+
+    /**
+     * Replaces local directory build cache.
+     */
+    void setLocal(DirectoryBuildCache local);
+
+    /**
+     * Replaces remote build cache.
+     */
+    void setRemote(@Nullable BuildCache remote);
```

Comment:
I wish we had some kind of lifecycle management for these configuration objects. Right now they stay mutable throughout the build. :/ Not something to fix here, but we are making the situation somewhat worse with these changes, and that makes me sad.